vm/src/types/slot.rs

```diff
@@ -85,6 +84,7 @@ pub struct PyTypeSlots {
 
     // The count of tp_members.
     pub member_count: usize,
+    pub number: PyNumberSlots,
```

Comment:
@xiaozhiyan Sorry for late comment 😂Could I ask you the reason why not to use `Option` for `PyNumberSlots`? If we want to know which type implements the Number protocol, shouldn't we verify all the fields in the Number methods?

Reply:
For a type to implement Number Protocol, basically it means the type implements a subset of Number Protocol methods.
I'm not sure what the scenario is to check whether a type has implemented "at least one Number Protocol method", instead of checking whether a type has implemented "certain specified method".
If it's really needed in such scenarios, I think we may be able to change this implementation to use `Option` without side effect. Otherwise, it may be better to keep it as is to avoid an additional layer of `Option` unwrapping.

Reply:
due to difference of AsNumber/PyNumberSlots implementation strategy, i guess PyNumberSlots always exists while AsNumber is not.

Reply:
According to current implementation,

- for embedded types, if they don't implement `AsNumber` trait, `extend_slots` will not be executed, and their `PyNumberSlots` will contain default method implementation, i.e. `None`

- for user defined types, if they implement any Number Protocol method, say, `__radd__`, the function will be saved in  corresponding field of `PyNumberSlots` for the type

- when trying to call certain Number Protocol method for a type, `PyNumberSlots` will be looked up, without considering whether the type has implemented `AsNumber` or not

Reply:
Ah I see. Thanks for detail explanation 😊